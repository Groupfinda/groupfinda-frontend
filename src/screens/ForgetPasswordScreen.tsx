import React from "react";
import { Layout, Text } from "@ui-kitten/components";
import { StyleSheet } from "react-native";
import { ForgetPasswordForm } from "../components/Auth";
import { ScrollView } from "react-native";
import { TransparentBackHeader } from "../components/common";
import { SafeAreaView } from 'react-native-safe-area-context'
type Props = {};

const ForgetPasswordScreen: React.FC<Props> = () => {
  return (
    <SafeAreaView style={styles.wrapperStyle}>
      <Layout style={styles.wrapperStyle}>
        <TransparentBackHeader />
        <ScrollView>
          <Layout style={styles.containerStyle}>
            <Layout style={styles.headerStyle}>
              <Text style={styles.textStyle} category="h3">
                Forgot your password?
            </Text>
              <Text category="s1" appearance="hint">
                Tell us your username and email and we will reset it for you.
            </Text>
            </Layout>
            <ForgetPasswordForm />
          </Layout>
        </ScrollView>
      </Layout>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  wrapperStyle: {
    flex: 1,
  },
  containerStyle: {
    flex: 1,
    alignItems: "center",
    paddingHorizontal: 40,
    paddingTop: 40,
  },
  headerStyle: {
    alignSelf: "center",
    marginTop: 40,
    marginBottom: 60,
  },

  textStyle: {
    alignSelf: "center",
  },
});

export default ForgetPasswordScreen;
