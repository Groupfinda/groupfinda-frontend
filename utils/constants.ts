export const eventCategories = [
  "Outdoors",
  "Adventure",
  "Technology",
  "Health",
  "Sports",
  "Fitness",
  "Learning",
  "Food & Drink",
  "Culture",
  "Music",
  "Social",
  "Games",
  "Arts",
  "Pets",
  "Hobbies",
  "Volunteer",
];
