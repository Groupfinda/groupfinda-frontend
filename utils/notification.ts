import Constants from 'expo-constants'
import * as Permissions from 'expo-permissions'
import { Platform } from "react-native";
import * as Notifications from 'expo-notifications'

export async function registerForPushNotificationsAsync() {
    let token;
    if (Constants.isDevice) {
        const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
            finalStatus = status;
        }

        if (finalStatus !== 'granted') {
            alert("Notifications allow us to send you updates on group matches and chat messages!")
            return
        }

        token = (await Notifications.getExpoPushTokenAsync()).data
        console.log(token)
    }

    if (Platform.OS === 'android') {
        Notifications.setNotificationChannelAsync('Groupfinda', {
            name: 'Groupfinda',
            importance: Notifications.AndroidImportance.MAX,
            vibrationPattern: [0, 250, 250, 250],
            lightColor: "#FF231F7C"
        })
    }

    return token
}

export type GroupfindaBody = {
    id: string;
    type: string;
}

export type AndroidNotificationData = GroupfindaBody
export type IOSNotificationData = {
    aps: {
        alert: {
            body: string;
            "launch-image": string;
            subtitle: string;
            title: string;
        }
        category: string;
        sound: string;
        "thread-id": string;
    }
    body: GroupfindaBody
}