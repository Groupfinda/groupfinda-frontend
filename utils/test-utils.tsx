import React from "react";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { myTheme } from "../custom-theme";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { MockedProvider, MockedResponse } from "@apollo/react-testing";
import { render, RenderAPI, RenderOptions } from "react-native-testing-library";

const AllProviders = (apolloMocks: MockedResponse[]) => ({
  children,
}: {
  children: JSX.Element;
}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...myTheme }}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          {children}
        </MockedProvider>
      </ApplicationProvider>
    </>
  );
};

type CustomRenderType = (
  component: React.ReactElement<any>,
  options?: RenderOptions,
  apolloProps?: MockedResponse[]
) => RenderAPI;

const customRender: CustomRenderType = (
  ui,
  options,
  apolloProps: MockedResponse[] = []
) => render(ui, { wrapper: AllProviders(apolloProps), ...options });

export * from "react-native-testing-library";
export { customRender as render };
