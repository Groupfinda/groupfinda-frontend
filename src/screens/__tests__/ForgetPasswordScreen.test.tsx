import React from "react";
import { render } from "../../../utils/test-utils";
import ForgetPasswordScreen from "../ForgetPasswordScreen";

test("renders UI correctly", () => {
  const tree = render(<ForgetPasswordScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders text components correctly", () => {
  const component = render(<ForgetPasswordScreen />);
  const header = component.getByText("Forgot your password?");
  const subheading = component.getByText(
    "Tell us your username and email and we will reset it for you."
  );
  expect(header.props.category).toBe("h3");
  expect(subheading.props.category).toBe("s1");
  expect(subheading.props.appearance).toBe("hint");
});
