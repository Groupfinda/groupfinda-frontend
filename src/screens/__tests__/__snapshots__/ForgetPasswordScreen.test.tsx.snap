// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
          Object {
            "backgroundColor": "transparent",
            "marginTop": 15,
            "position": "absolute",
            "zIndex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "zIndex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#222B45"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "tintColor": "#222B45",
                    "width": 24,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={4280429381}
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={4278190080}
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    fill={4278190080}
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGRect
                      fill={4278190080}
                      fillOpacity={1}
                      fillRule={1}
                      height="24"
                      matrix={
                        Array [
                          6.123233995736766e-17,
                          1,
                          -1,
                          6.123233995736766e-17,
                          24,
                          0,
                        ]
                      }
                      opacity={0}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M19 11H7.14l3.63-4.36a1 1 0 1 0-1.54-1.28l-5 6a1.19 1.19 0 0 0-.09.15c0 .05 0 .08-.07.13A1 1 0 0 0 4 12a1 1 0 0 0 .07.36c0 .05 0 .08.07.13a1.19 1.19 0 0 0 .09.15l5 6A1 1 0 0 0 10 19a1 1 0 0 0 .64-.23 1 1 0 0 0 .13-1.41L7.14 13H19a1 1 0 0 0 0-2z"
                      fill={4278190080}
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View />
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "zIndex": 1,
            },
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            },
          ]
        }
      />
    </View>
    <RCTScrollView>
      <View>
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "paddingHorizontal": 40,
                "paddingTop": 40,
              },
            ]
          }
        >
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {
                  "alignSelf": "center",
                  "marginBottom": 60,
                  "marginTop": 40,
                },
              ]
            }
          >
            <Text
              appearance="default"
              category="h3"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 30,
                    "fontWeight": "800",
                  },
                  Object {
                    "alignSelf": "center",
                  },
                ]
              }
            >
              Forgot your password?
            </Text>
            <Text
              appearance="hint"
              category="s1"
              style={
                Array [
                  Object {
                    "color": "#8F9BB3",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "600",
                  },
                  undefined,
                ]
              }
            >
              Tell us your username and email and we will reset it for you.
            </Text>
          </View>
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {
                  "alignSelf": "stretch",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginBottom": 5,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "color": "#8F9BB3",
                        "fontFamily": "System",
                        "fontSize": 12,
                        "fontWeight": "800",
                        "marginBottom": 4,
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ],
                  ]
                }
              >
                Username
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F7F9FC",
                      "borderColor": "#E4E9F2",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "minHeight": 40,
                      "paddingHorizontal": 8,
                      "paddingVertical": 7,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={24}
                    bbWidth={24}
                    fill="#8F9BB3"
                    focusable={false}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "height": 24,
                          "marginHorizontal": 8,
                          "tintColor": "#8F9BB3",
                          "width": 24,
                        },
                        Object {
                          "opacity": 1,
                        },
                        Object {
                          "flex": 0,
                          "height": 24,
                          "width": 24,
                        },
                      ]
                    }
                    vbHeight={24}
                    vbWidth={24}
                  >
                    <RNSVGGroup
                      fill={4287601587}
                      fillOpacity={1}
                      fillRule={1}
                      font={Object {}}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    >
                      <RNSVGGroup
                        fill={4278190080}
                        fillOpacity={1}
                        fillRule={1}
                        font={Object {}}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={Array []}
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth={1}
                        vectorEffect={0}
                        x={0}
                        y={0}
                      >
                        <RNSVGGroup
                          fill={4278190080}
                          fillOpacity={1}
                          fillRule={1}
                          font={Object {}}
                          matrix={
                            Array [
                              1,
                              0,
                              0,
                              1,
                              0,
                              0,
                            ]
                          }
                          opacity={1}
                          originX={0}
                          originY={0}
                          propList={Array []}
                          rotation={0}
                          scaleX={1}
                          scaleY={1}
                          skewX={0}
                          skewY={0}
                          stroke={null}
                          strokeDasharray={null}
                          strokeDashoffset={null}
                          strokeLinecap={0}
                          strokeLinejoin={0}
                          strokeMiterlimit={4}
                          strokeOpacity={1}
                          strokeWidth={1}
                          vectorEffect={0}
                          x={0}
                          y={0}
                        >
                          <RNSVGRect
                            fill={4278190080}
                            fillOpacity={1}
                            fillRule={1}
                            height="24"
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={0}
                            originX={0}
                            originY={0}
                            propList={Array []}
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            width="24"
                            x={0}
                            y={0}
                          />
                          <RNSVGPath
                            d="M12 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4zm0-6a2 2 0 1 1-2 2 2 2 0 0 1 2-2z"
                            fill={4278190080}
                            fillOpacity={1}
                            fillRule={1}
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={1}
                            originX={0}
                            originY={0}
                            propList={Array []}
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            x={0}
                            y={0}
                          />
                          <RNSVGPath
                            d="M12 13a7 7 0 0 0-7 7 1 1 0 0 0 2 0 5 5 0 0 1 10 0 1 1 0 0 0 2 0 7 7 0 0 0-7-7z"
                            fill={4278190080}
                            fillOpacity={1}
                            fillRule={1}
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={1}
                            originX={0}
                            originY={0}
                            propList={Array []}
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            x={0}
                            y={0}
                          />
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
                <TextInput
                  allowFontScaling={true}
                  appearance="default"
                  autoCapitalize="none"
                  autoCorrect={false}
                  blurOnSubmit={false}
                  editable={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Username"
                  placeholderTextColor="#8F9BB3"
                  rejectResponderTermination={true}
                  size="medium"
                  status="basic"
                  style={
                    Array [
                      Object {
                        "color": "#222B45",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "normal",
                        "marginHorizontal": 8,
                      },
                      Object {
                        "flexBasis": "auto",
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                      null,
                      undefined,
                    ]
                  }
                  textContentType="username"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                  ]
                }
              />
            </View>
            <View
              style={Object {}}
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "color": "#8F9BB3",
                        "fontFamily": "System",
                        "fontSize": 12,
                        "fontWeight": "800",
                        "marginBottom": 4,
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ],
                  ]
                }
              >
                Email
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F7F9FC",
                      "borderColor": "#E4E9F2",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "minHeight": 40,
                      "paddingHorizontal": 8,
                      "paddingVertical": 7,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={24}
                    bbWidth={24}
                    fill="#8F9BB3"
                    focusable={false}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "height": 24,
                          "marginHorizontal": 8,
                          "tintColor": "#8F9BB3",
                          "width": 24,
                        },
                        Object {
                          "opacity": 1,
                        },
                        Object {
                          "flex": 0,
                          "height": 24,
                          "width": 24,
                        },
                      ]
                    }
                    vbHeight={24}
                    vbWidth={24}
                  >
                    <RNSVGGroup
                      fill={4287601587}
                      fillOpacity={1}
                      fillRule={1}
                      font={Object {}}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    >
                      <RNSVGGroup
                        fill={4278190080}
                        fillOpacity={1}
                        fillRule={1}
                        font={Object {}}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={Array []}
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth={1}
                        vectorEffect={0}
                        x={0}
                        y={0}
                      >
                        <RNSVGGroup
                          fill={4278190080}
                          fillOpacity={1}
                          fillRule={1}
                          font={Object {}}
                          matrix={
                            Array [
                              1,
                              0,
                              0,
                              1,
                              0,
                              0,
                            ]
                          }
                          opacity={1}
                          originX={0}
                          originY={0}
                          propList={Array []}
                          rotation={0}
                          scaleX={1}
                          scaleY={1}
                          skewX={0}
                          skewY={0}
                          stroke={null}
                          strokeDasharray={null}
                          strokeDashoffset={null}
                          strokeLinecap={0}
                          strokeLinejoin={0}
                          strokeMiterlimit={4}
                          strokeOpacity={1}
                          strokeWidth={1}
                          vectorEffect={0}
                          x={0}
                          y={0}
                        >
                          <RNSVGRect
                            fill={4278190080}
                            fillOpacity={1}
                            fillRule={1}
                            height="24"
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={0}
                            originX={0}
                            originY={0}
                            propList={Array []}
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            width="24"
                            x={0}
                            y={0}
                          />
                          <RNSVGPath
                            d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm-.67 2L12 10.75 5.67 6zM19 18H5a1 1 0 0 1-1-1V7.25l7.4 5.55a1 1 0 0 0 .6.2 1 1 0 0 0 .6-.2L20 7.25V17a1 1 0 0 1-1 1z"
                            fill={4278190080}
                            fillOpacity={1}
                            fillRule={1}
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={1}
                            originX={0}
                            originY={0}
                            propList={Array []}
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            x={0}
                            y={0}
                          />
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
                <TextInput
                  allowFontScaling={true}
                  appearance="default"
                  autoCapitalize="none"
                  autoCorrect={false}
                  editable={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Email"
                  placeholderTextColor="#8F9BB3"
                  rejectResponderTermination={true}
                  size="medium"
                  status="basic"
                  style={
                    Array [
                      Object {
                        "color": "#222B45",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "normal",
                        "marginHorizontal": 8,
                      },
                      Object {
                        "flexBasis": "auto",
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                      null,
                      undefined,
                    ]
                  }
                  textContentType="emailAddress"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                  ]
                }
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#008EED",
                  "borderColor": "#008EED",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 40,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                Reset my password
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
